plugins/inputs/modbus/modbus.go

```diff
@@ -642,11 +658,22 @@ func (m *Modbus) Gather(acc telegraf.Accumulator) error {
 	}
 
 	timestamp := time.Now()
-	err := m.getFields()
-	if err != nil {
-		disconnect(m)
-		m.isConnected = false
-		return err
+	for retry := 0; retry <= m.Retries; retry += 1 {
+		timestamp = time.Now()
+		err := m.getFields()
+		if err != nil {
+			mberr, ok := err.(*mb.ModbusError)
+			if ok && mberr.ExceptionCode == mb.ExceptionCodeServerDeviceBusy && retry < m.Retries {
+				log.Printf("I! [inputs.modbus] device busy! Retrying %d more time(s)...", m.Retries-retry)
+				continue
+			}
+			disconnect(m)
+			m.isConnected = false
+			return err
+		}
+		// Reading was successful, leave the retry loop
+		break
+		time.Sleep(m.RetriesWaitTime.Duration)
```

Comment:
This sleep is now unreachable (and it's breaking the build), I think you want to move it to line 667 after the log.Printf and before the continue?

Reply:
Done with the commit(s) below.